ibis/backends/duckdb/__init__.py

```diff
@@ -1797,3 +1903,30 @@ def _pyarrow_rbr(source, table_name, _conn, **_: Any):
     # Ensure the reader isn't marked as started, in case the name is
     # being overwritten.
     _conn._record_batch_readers_consumed[table_name] = False
+
+
+def _attach_name(path_or_url: str | Path) -> str:
+    """Return the name to use when attaching a database."""
+    url = urllib.parse.urlparse(str(path_or_url))
+    final = Path(url.path).stem
+    if not final:
+        raise ValueError(
+            f"Could not determine a name for the database from {path_or_url!r}. "
+            "Please provide a name explicitly using the `name` parameter."
+        )
+    return final
+
+
+for inp, expected in [
```

Comment:
This was me avoiding importing this private function into the test module. We could also just not test this at all, and call it an implementation detail/undefined behavior. Or I could put it in the test module. Or we could adjust the API of the .attach() method to not return the attached name and have duckdb name it whatever it wants opaquely to the user, which would allow us to just remove all this code.